src/test/java/com/drajer/test/ITPageRetry.java

```diff
@@ -0,0 +1,152 @@
+
+  public ITPageRetry(
+      String testCaseId,
+      Map<String, String> testData,
+      List<Map<String, String>> validateFields,
+      Map<String, ?> resourceMapping,
+      Map<String, ?> otherMapping) {
+
+    this.testCaseId = testCaseId;
+    this.testData = testData;
+    this.fieldsToValidate = validateFields;
+    this.allResourceMapping = resourceMapping;
+    this.allOtherMapping = otherMapping;
+  }
+
+  private static final Logger logger = LoggerFactory.getLogger(ITPageRetry.class);
+  private static String systemLaunchPayload;
+  private LaunchDetails launchDetails;
+  private PatientExecutionState state;
+  WireMockHelper stubHelper;
+
+  @Before
+  public void launchTestSetUp() throws Throwable {
+    logger.info("Executing Test: {}", testCaseId);
+    tx = session.beginTransaction();
+
+    // Data Setup
+    createClientDetails(testData.get("ClientDataToBeSaved"));
+    systemLaunchPayload = getSystemLaunchPayload(testData.get("SystemLaunchPayload"));
+    session.flush();
+    tx.commit();
+
+    stubHelper = new WireMockHelper(wireMockServer, wireMockHttpPort);
+    logger.info("Creating WireMock stubs..");
+    stubHelper.stubResources(allResourceMapping);
+    stubHelper.stubAuthAndMetadata(allOtherMapping);
+    mockRestApi();
+  }
+
+  @Parameterized.Parameters(name = "{0}")
+  public static Collection<Object[]> data() {
+
+    List<TestDataGenerator> testDataGenerator = new ArrayList<>();
+    testDataGenerator.add(new TestDataGenerator("test-yaml/fhirRetryPageTest.yaml"));
+
+    int totalTestCount = 0;
+    for (TestDataGenerator testData : testDataGenerator) {
+      totalTestCount = totalTestCount + testData.getAllTestCases().size();
+    }
+
+    Object[][] data = new Object[totalTestCount][5];
+
+    int count = 0;
+    for (TestDataGenerator testData : testDataGenerator) {
+      Set<String> testCaseSet = testData.getAllTestCases();
+      for (String testCase : testCaseSet) {
+        data[count][0] = testCase;
+        data[count][1] = testData.getTestCaseByID(testCase).getTestData();
+        data[count][2] = testData.getValidate(testCase);
+        data[count][3] = testData.getResourceMappings(testCase);
+        data[count][4] = testData.getOtherMappings(testCase);
+        count++;
+      }
+    }
+    return Arrays.asList(data);
+  }
+
+  @Test
+  public void testPage() throws Exception {
+
+    ResponseEntity<String> response = invokeSystemLaunch(testCaseId, systemLaunchPayload);
+
+    assertEquals(HttpStatus.ACCEPTED, response.getStatusCode());
+    assertTrue(response.getBody().contains("App is launched successfully"));
+
+    logger.info("Received success response, waiting for EICR generation.....");
+    Eicr createEicr = getCreateEicrDocument();
+    wireMockServer.verify(
+        15,
+        getRequestedFor(
+            urlEqualTo(
+                "/FHIR/Observation?patient=12742571&category=laboratory&-pageContext=10065315_11316911_12742571_1_1&-pageDirection=NEXT")));
```

Comment:
Maybe you can validate like below to avoid occasional failures. Same for other tests.
```suggestion
    wireMockServer.verify(
        moreThanOrExactly(15),
        getRequestedFor(
            urlEqualTo(
                "/FHIR/Observation?patient=12742571&category=laboratory&-pageContext=10065315_11316911_12742571_1_1&-pageDirection=NEXT")));
```